core/dnsserver/server-grpc.go

```diff
@@ -0,0 +1,152 @@
+package dnsserver
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	// TODO(miek): can't have these cross refs, pb needs to move here, or in the top-level dir
+	"github.com/coredns/coredns/middleware/proxy/pb"
+
+	"github.com/miekg/dns"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+)
+
+// GRPCServer represents an instance of a DNS-over-gRPC.
+type GRPCServer struct {
```

Comment:
No. Same holds for TLS server, but tLS will make `go lint` complain...hence exported. I'll reverse the name serverTLS, servergRPC.